compiler/src/iree/compiler/GlobalOptimization/QuantizedConvToConv.cpp

```diff
@@ -163,81 +172,206 @@ struct QuantizedConvToConv
+          dilations, inputNDimIndex, inputHDimIndex, inputWDimIndex,
+          inputCDimIndex, filterHDimIndex, filterWDimIndex);
+    }
 
-      // Materialize a length-1 dimension at the end of the summation.
-      SmallVector<ReassociationExprs> reassociationMap(3);
-      for (int i = 0; i < 3; i++)
-        reassociationMap[i].push_back(builder.getAffineDimExpr(i));
-      reassociationMap.back().push_back(builder.getAffineDimExpr(3));
+    // Apply the final update that occurs when there are multiple zero-points.
+    if (!iZpIsZero && !fZpIsZero) {
+      newConv = applyFinalZeroPointUpdate(builder, newConv, filter, iZp, fZp,
+                                          accETy, filterHDimIndex,
+                                          filterWDimIndex, filterCDimIndex);
+    }
 
-      auto expandTy =
-          RankedTensorType::get({inputTy.getDimSize(0), inputTy.getDimSize(1),
-                                 inputTy.getDimSize(2), 1},
-                                accETy);
-      inputSum = builder.create<tensor::ExpandShapeOp>(expandTy, inputSum,
-                                                       reassociationMap);
+    rewriter.replaceOp(op, newConv);
+    return success();
+  }
 
-      llvm::SmallVector<int64_t> poolDims;
-      llvm::SmallVector<Value> poolDynDims;
-      GetDynamicDym(builder, poolDims, poolDynDims, inputSum, 0);
-      GetDynamicDym(builder, poolDims, poolDynDims, newConv, 1);
-      GetDynamicDym(builder, poolDims, poolDynDims, newConv, 2);
-      GetDynamicDym(builder, poolDims, poolDynDims, inputSum, 3);
+private:
+  bool isConvNHWC(Conv2DQOpType op) const {
+    if (auto convOp = dyn_cast<linalg::Conv2DNhwcHwcfQOp>(&op)) {
+      return true;
+    } else if (auto convOp = dyn_cast<linalg::Conv2DNchwFchwQOp>(&op)) {
+      return false;
+    } else {
+      llvm_unreachable("Unsupported Conv2DOp type!");
+    }
+  }
 
-      // Perform a sum-pooling operation across the kernel width and height.
-      auto poolTy = RankedTensorType::get(poolDims, accETy);
-      Value poolTensor = emptyZero(builder, poolTy, poolDynDims);
+  void getConv2DOpIndices(bool isNHWC, int &filterFDimIndex,
+                          int &filterCDimIndex, int &filterHDimIndex,
+                          int &filterWDimIndex, int &inputNDimIndex,
+                          int &inputHDimIndex, int &inputWDimIndex,
+                          int &inputCDimIndex) const {
+    if (isNHWC) {
+      // hwcf
+      filterHDimIndex = 0;
+      filterWDimIndex = 1;
+      filterCDimIndex = 2;
+      filterFDimIndex = 3;
+      // nhwc
+      inputNDimIndex = 0;
+      inputHDimIndex = 1;
+      inputWDimIndex = 2;
+      inputCDimIndex = 3;
+    } else {
+      // fchw
+      filterFDimIndex = 0;
+      filterCDimIndex = 1;
+      filterHDimIndex = 2;
+      filterWDimIndex = 3;
+      // nchw
+      inputNDimIndex = 0;
+      inputCDimIndex = 1;
+      inputHDimIndex = 2;
+      inputWDimIndex = 3;
+    }
+  }
```

Comment:
Double check with `inferConvolutionDims()` and `ConvolutionDimensions class` per @Max191 pointed out